cockatrice/src/window_main.cpp

```diff
@@ -288,6 +293,87 @@ void MainWindow::actAbout()
     ));
 }
 
+void MainWindow::actUpdate()
+{
+    //Request the newest builds
+    QUrl versionUrl(LATEST_FILES_URL);
+    response = netMan.get(QNetworkRequest(versionUrl));
+    connect(response, SIGNAL(finished()),
+            this, SLOT(fileListFinished()));
+}
+
+void MainWindow::fileListFinished() {
+    //Parse the version results into a JSON array
+    QString strReply = (QString) response->readAll();
+    QJsonDocument jsonResponse = QJsonDocument::fromJson(strReply.toUtf8());
+    QJsonArray json_array = jsonResponse.array();
+
+    //From this array we intend to find the correct downloadUrl, if needed
+    QString downloadUrl;
+    bool correctVersion = false, needToUpdate = false;
+    QString dateStr;
+    int downloadSize;
+
+    //Iterate over the builds
+    foreach (const QJsonValue &value, json_array)
```

Comment:
Variable names like `value` obscure the intent of the code. How about `versionMetadata`? With a name like the comment becomes unnecessary. Similarly, `json_array` gives me no information since the type already has that info; maybe `allBintrayVersions`?